ConsoleProject/src/services/TextureConverter.cs

```diff
@@ -75,8 +75,39 @@ public class TextureConverter {
 
             return texture;
         }
-
+        
         private Bitmap GetBitmapFromPalette(Texture texture) {
+            Bitmap bmp = new Bitmap(texture.Width, texture.Height);
+            int row = 0;
+            int bitsPerPixelMultiplier = 8 / texture.BitsPerPixel;
+
+            List<System.Drawing.Color> colors = new List<System.Drawing.Color>();
+            
+            for (int c = 0; c < texture.Palette.Length / 4; c++) {
```

Comment:
Move this to a private method, with return type: List<Colors>.
As parameter pass the texture.Palette array.

On this level call this only:
texture.colors.AddAll( privateMethod );